docs/hugo/content/architecture-decision-records/ADR-2022-02-Adoption-Policy.md

```diff
@@ -0,0 +1,127 @@
+     by ASO at all. It was created and managed some other way (portal, terraform, az-cli, etc).
+3. **Users must be able to instruct the operator to skip deletion of the backing resource:**
+   This is useful for scenarios such as moving ownership of resources from ASO in cluster A to ASO in cluster B.
+   This is enabled by `reconcile-policy`.
+
+### Open question: what should the default `reconcile-policy` be for resources which are adopted?
+
+There are two main options.
+
+#### Option 1: The default `reconcile-policy` for adopted resources is the same as for resources ASO created.
+
+ASO automatically adopts, updates, and will delete backing Azure resources as normal unless the user
+explicitly sets `reconcile-policy`.
+
+**Advantages:**
+- It's easy to explain what the behavior is. Simplistic users don't have to understand `reconcile-policy` (they won't
+  ever see it unless they set it themselves).
+
+**Disadvantages:**
+- Users have no good way to know their resource existed previously. They may `kubectl delete` it without realizing that
+  other things depend on this resource. Deletion (and some mutation) is destructive in Azure so lost data, etc cannot 
+  easily be recovered.
+
+**Thoughts/comments:**
+
+This sort-of matches the behavior of ARM templates and SDKs. If you call the `DELETE` API the resource is gone.
+
+Some discussion of this topic has happened over in [ACK](https://github.com/aws-controllers-k8s/community/issues/82).
+
+[Crossplane](https://github.com/crossplane/crossplane/issues/2590) doesn't support any notion of `reconcile-policy` at all.
+`kubectl delete` always deletes the backing resource no matter what.
+
+[GCP](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources#creating_a_resource) does it this way:
+> When you create a resource, Config Connector creates the resource if it doesn't exist. If a Google Cloud resource 
+> already exists with the same name, then Config Connector acquires the resource and manages it.
+
+GCP also [supports](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources) a 
+`reconcile-policy`-like annotation, so doing it like this would basically be doing it exactly like GCP.
+
+Kuberentes has a similar concept for `StorageClasses`/`PersistentVolumes` called 
+[reclaimPolicy](https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy), whose default
+seems to be `Delete`.
+
+#### Option 2: The default `reconcile-policy` for adopted resource is `abandon-on-delete`.
+
+ASO detects that a resource previously existed and sets `reconcile-policy` automatically to `abandon-on-delete`.
+A new option in the ASO config `adoption-policy` would control this default so users could opt into the Option 1 style
+discussed above if they wanted (but it's not the default behavior).
+
+**Advantages:**
+- More flexibility. Users can choose which behavior they want when deploying ASO. 
```

Comment:
Hmm correct me if I'm wrong, but I don't see this as a distinctive advantage to this approach. Users should be just as flexible with the other approach as they can actively choose and set the reconcile-policy. This would just be to attempt a better default, an attempt in guessing what the user actually wants without them actively setting it.

Reply:
What I was getting at with flexibility here was that, if users wanted the operator to autodetect adopted resources, it could. If they didn't want that, they can just set `adoption-policy` and get the option1 behavior. We'd basically be letting them choose which behavior they want.

Whereas if we just go with option1 and no `adoption-policy` at the operator level then the users can't have the operator "know" for them, they'll have to know and annotate things accordingly.